include/pybind11/detail/common.h

```diff
@@ -89,6 +89,16 @@
 #  endif
 #endif
 
+// Exceptions can't be marked with __declspec(dllexport) in MinGW
+#if !defined(PYBIND11_EXPORT)
```

Comment:
#if !defined(PYBIND11_EXPORT_EXCEPTION)
?

Reply:
That's what I get for doing this right before going to sleep...